File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
	int cnt = 0;
	for (int i = 0; i < s.size(); i++)
	{
		if (!(s[i] >= 'a' && s[i] <= 'z') && !(s[i] >= 'A' && s[i] <= 'Z'))
		{
			answer.push_back(s[i]);
			cnt = 0;
			continue;
		}

		if (cnt % 2 == 0)
		{
			if (s[i] >= 'a' && s[i] <= 'z')
				answer.push_back(s[i] - 32);
			else answer.push_back(s[i]);
		}
		else
		{
			if (s[i] >= 'a' && s[i] <= 'z')
				answer.push_back(s[i]);
			else answer.push_back(s[i] + 32);
		}

		cnt++;
	}
    return answer;
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.cs
public class Solution {
    public string solution(string s) {
        string answer = "";
        int n = 0;
        foreach(char c in s)
        {
            if(c == ' ')
            {
                n = 0;
                answer += " ";
                continue;
            }
            
            if(n%2 == 0)
            {
                answer += char.ToUpper(c);
                n++;
            }
            else 
            {
                answer += char.ToLower(c);
                n++;
            }
            
        }
        return answer;
    }
}